eng/ci/templates/official/jobs/setup-benchmark-agents.yml

```diff
@@ -0,0 +1,103 @@
+parameters:
+- name: functionAppName
+  type: string
+- name: agentName
+  type: string
+jobs:
+- job: ${{ parameters.functionAppName }}
+
+  variables:
+    agentName: ${{ parameters.agentName }}
+    functionApp: ${{ parameters.functionAppName }}
+    functionAppOutputPath: $(Build.ArtifactStagingDirectory)/FunctionApps/$(functionApp)
+
+  steps:
+
+  - template: /eng/ci/templates/install-dotnet.yml@self
+
+  - script: dotnet tool install -g Microsoft.Crank.Agent --version "0.2.0-*"
+    displayName: Install crank agent
+
+  - pwsh: Start-Process powershell -ArgumentList '-NoExit', '-Command', 'crank-agent'
+    displayName: Start crank agent
+
+  - pwsh: |
+      dotnet --info
+    displayName: Print .NET info
+
+  - task: CopyFiles@2
+    displayName: Copy benchmark apps to temp location
+    inputs:
+      SourceFolder: '$(Build.SourcesDirectory)/test/Performance/Apps'
+      Contents: '**/*'
+      TargetFolder: '$(Build.ArtifactStagingDirectory)/PerformanceTestApps'
+      CleanTargetFolder: true
+
+  - task: DotNetCoreCLI@2
+    displayName: Publish benchmark app
+    inputs:
+      command: publish
+      publishWebProjects: false
+      zipAfterPublish: false
+      modifyOutputPath: false
+      projects: '$(Build.ArtifactStagingDirectory)/PerformanceTestApps/$(functionApp)/HelloHttp.csproj'
+      arguments: -c Release -o $(functionAppOutputPath) -f net9.0 -v n
+      workingDirectory: $(Build.ArtifactStagingDirectory)/PerformanceTestApps/$(functionApp)
+
+  - script: |
+      netsh advfirewall firewall add rule name="Open Port 5000" dir=in action=allow protocol=TCP localport=5000
+      netsh advfirewall firewall add rule name="Open Port 5010" dir=in action=allow protocol=TCP localport=5010
+    displayName: Open port 5000 and 5010
+
+  - task: PowerShell@2
+    displayName: Get Agent IP Address
+    inputs:
+      targetType: 'inline'
+      script: |
+        $ipAddress = (Test-Connection -ComputerName (hostname) -Count 1).IPv4Address.IPAddressToString
+        Write-Host "IP Address: $ipAddress"
+        Write-Host "##vso[task.setvariable variable=agentIp]$ipAddress"
+
+  - task: AzureCLI@2
```

Comment:
We need to share the agent info(IP address) to the other job(agent) so that It can communicate with this agent (communication from load to application). Job output variable won't work as it is available only after the job completes. We want the job(agent) running.

Reply:
Interesting - there isn't a more pipelines-native way to do this?